include/cooperative_perception/ctrv_model.hpp

```diff
@@ -1,21 +1,141 @@
+/*
+ * Copyright 2022 Leidos
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 #ifndef COOPERATIVE_PERCEPTION_CTRV_MODEL_HPP
 #define COOPERATIVE_PERCEPTION_CTRV_MODEL_HPP
 
+#include <boost/container_hash/hash.hpp>
+#include <boost/math/special_functions/next.hpp>
+#include <functional>
+#include <Eigen/Dense>
 #include <units.h>
 #include "cooperative_perception/units.hpp"
 
 namespace cooperative_perception
 {
-
 struct CtrvState
 {
   units::length::meter_t position_x;
   units::length::meter_t position_y;
   units::velocity::meters_per_second_t velocity;
   units::angle::radian_t yaw;
   units::angular_velocity::radians_per_second_t yaw_rate;
+
+  static constexpr auto kNumVars{ 5 };
+
+  static inline auto fromEigenVector(const Eigen::Vector<float, kNumVars>& vec) noexcept -> CtrvState
+  {
+    return CtrvState{ .position_x{ units::length::meter_t{ vec(0) } },
+                      .position_y{ units::length::meter_t{ vec(1) } },
+                      .velocity{ units::velocity::meters_per_second_t{ vec(2) } },
+                      .yaw{ units::angle::radian_t{ vec(3) } },
+                      .yaw_rate{ units::angular_velocity::radians_per_second_t{ vec(4) } } };
+  }
 };
 
+inline auto operator+=(CtrvState& lhs, const CtrvState& rhs) -> CtrvState&
+{
+  lhs.position_x += rhs.position_x;
+  lhs.position_y += rhs.position_y;
+  lhs.velocity += rhs.velocity;
+  lhs.yaw += rhs.yaw;
+  lhs.yaw_rate += rhs.yaw_rate;
+
+  return lhs;
+}
+
+inline auto operator-=(CtrvState& lhs, const CtrvState& rhs) -> CtrvState&
+{
+  lhs.position_x -= rhs.position_x;
+  lhs.position_y -= rhs.position_y;
+  lhs.velocity -= rhs.velocity;
+  lhs.yaw -= rhs.yaw;
+  lhs.yaw_rate -= rhs.yaw_rate;
+
+  return lhs;
+}
+
+inline auto operator==(const CtrvState& lhs, const CtrvState& rhs) -> bool
+{
+  return lhs.position_x == rhs.position_x && lhs.position_y == rhs.position_y && lhs.velocity == rhs.velocity &&
+         lhs.yaw == rhs.yaw && lhs.yaw_rate == rhs.yaw_rate;
+}
+
+inline auto operator+(CtrvState lhs, const CtrvState& rhs) -> CtrvState
+{
+  lhs += rhs;
+  return lhs;
+}
+
+inline auto operator-(CtrvState lhs, const CtrvState& rhs) -> CtrvState
+{
+  lhs -= rhs;
+  return lhs;
+}
+
+using CtrvStateCovariance = Eigen::Matrix<float, 5, 5>;
```

Comment:
The matrix dimensions should use `CtrvState::kNumVars` to avoid size mismatches.

Reply:
Would this be solved by reusing `static constexpr auto kNumVars{ 5 };` from further up the file? So:
`Eigen::Matrix<float, kNumVars, kNumVars>`

Reply:
Makes sense - I will make changes tomorrow